api/src/pcapi/scripts/unlock_users/main.py

```diff
@@ -0,0 +1,51 @@
+import argparse
+import logging
+
+from pcapi.app import app
+from pcapi.core.finance.enum import DepositType
+from pcapi.core.finance.models import Deposit
+from pcapi.core.subscription.api import activate_beneficiary_if_no_missing_step
+from pcapi.core.users.models import User
+from pcapi.core.users.models import UserRole
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+app.app_context().push()
+
+
+def main(not_dry: bool) -> None:
+    query = (
+        db.session.query(User)
+        .join(User.deposits)
+        .filter(
+            User.has_beneficiary_role,
+            Deposit.type == DepositType.GRANT_15_17,
+            Deposit.expirationDate > db.func.now(),
+        )
+    )
+    users_to_update = query.all()
+    logger.info("Found %s users to update", len(users_to_update))
+    for user in users_to_update:
+        logger.info("Updating user %s", user.id)
+        user.roles = [UserRole.UNDERAGE_BENEFICIARY]
```

Comment:
```suggestion
        user.remove_beneficiary_role()
```
est-ce qu'il existe des utilisateurs bloqués avec le role `BENEFICIARY` mais sans aucun `Deposit` ?

Reply:
Apparement non 👍 
![image](https://github.com/user-attachments/assets/718f0db1-3759-4f51-9e58-72fc6d7c01c4)